### PR TITLE
Remove duplicate tests from System.Text.Encoding.Extensions

### PR DESCRIPTION
**Pull request by hughbe**
- Only added tests for GetMaxByteCount/GetMaxByteCount and
encoding/decoding nothing

In a future PR, I will move the non-duplicated tests to the new test files - this way its less confusing and easier for you to review.

Contributes to #8114

/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding.Extensions/tests/ASCII.cs [hunk 1]
```diff
@@ -15,142 +15,26 @@ public static class ASCII
...
-            s_encodingUtil_ASCII.GetByteCountTest(String.Empty, 0, 0, 0);
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest(String.Empty, 0, 1, 0));
-            Assert.Throws<ArgumentNullException>(() => s_encodingUtil_ASCII.GetByteCountTest((String)null, 0, 0, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest("abc", -1, 0, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest("abc", 0, -1, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest("abc", -1, -1, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest("abc", 1, -1, 0));
-            s_encodingUtil_ASCII.GetByteCountTest("abc", 3, 0, 0);
-            Assert.Throws<ArgumentOutOfRangeException>(() => s_encodingUtil_ASCII.GetByteCountTest("abc", 3, 1, 0));
             s_encodingUtil_ASCII.GetByteCountTest("abc", 2, 1, 1);
```
Comment (tarekgh) [inline src/System.Text.Encoding.Extensions/tests/ASCII.cs]: in general, why we still keeping such tests? I am seeing we still keeping many of them. is that because the tests on System.Text.Encoding is not covering such cases yet? or is there any other reason?

Comment (hughbe) [inline src/System.Text.Encoding.Extensions/tests/ASCII.cs]: yeah we don't have any cases currently where `index + count == string.Length` over in the modern files

Comment (tarekgh) [inline src/System.Text.Encoding.Extensions/tests/ASCII.cs]: can we try support such case with this change? we'll just do one round clean up. if you prefer do it in separate PR that fine too.

Comment (hughbe) [inline src/System.Text.Encoding.Extensions/tests/ASCII.cs]: I'm planning to move all of the non-duplicated tests to the new files in a future PR if that's OK?

Comment (tarekgh) [inline src/System.Text.Encoding.Extensions/tests/ASCII.cs]: sure. this will be fine too.

**Comment by tarekgh** [comment]
test Innerloop Windows_NT Debug Build and Test please

**Comment by tarekgh** [comment]
LGTM. Thanks
